### ISSUE TITLE
[RFC] move collapsible_if from style to pedantic group

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang-nursery/rust-clippy#3228]
What are the opinions on moving `collapsible_if` to the pedantic group?

I agree that it is not useful to have super-deep nested if-chains but having an if with a lot of conditions chained together also decreases readability in my opinion.
Sometimes the branches are explicitly commented which the lint ignores.

````rust
fn main () {
    let c: bool = true;
    let b: bool = false;
    if c {
    // if c is true, we know everything is alright
        if b {
        // this should never happen
            panic!("welp!");
        } 
    }
}
````

There are many repos which already opted out of the lint:
https://github.com/search?q=clippy+allow%28collapsible_if%29&type=Code

I suggest moving the lint to pedantic group.
Comments?

**Comment by flip1995** [comment]
Maybe the enhancement suggested here #798 would be an alternative.

**Comment by Manishearth** [comment]
Yeah, I think it's worth investigating _why_ people disable it, I think this is a good lint.

THe comment thing, as well as a limit on the nesting depth, may help.

**Comment by matthiaskrgr** [comment]
If comments were taken into account this would definitely improve the situation, yet I think this lint is probably better off being an (optional) rustfmt formatting change instead of a code style warning.

**Comment by matthiaskrgr** [comment]
I'm closing this since the pr that makes the lint comment-sensitive has been merged! :)
